kanidm_client/tests/proto_v1_test.rs

```diff
@@ -987,6 +987,80 @@ fn test_server_rest_webauthn_mfa_auth_lifecycle() {
     });
 }
 
+#[test]
+fn test_server_rest_oauth2_basic_lifecycle() {
+    run_test(|rsclient: KanidmClient| {
+        let res = rsclient.auth_simple_password("admin", ADMIN_TEST_PASSWORD);
+        assert!(res.is_ok());
+
+        // List, there are non.
+        let initial_configs = rsclient
+            .idm_oauth2_rs_list()
+            .expect("Failed to retrieve oauth2 configs");
+
+        assert!(initial_configs.is_empty());
+
+        // Create a new oauth2 config
+        rsclient
+            .idm_oauth2_rs_basic_create("test_integration", "https://demo.example.com")
+            .expect("Failed to create oauth2 config");
+
+        // List, there is what we created.
+        let initial_configs = rsclient
+            .idm_oauth2_rs_list()
+            .expect("Failed to retrieve oauth2 configs");
+
+        assert!(initial_configs.len() == 1);
+
+        // Get the value. Assert we have oauth2_rs_basic_secret,
+        // but can NOT see the token_secret.
+        let oauth2_config = rsclient
+            .idm_oauth2_rs_get("test_integration")
+            .ok()
+            .flatten()
+            .expect("Failed to retrieve test_integration config");
```

Comment:
So oauth_rs_basic_secret is for authenticating the resource server? How does that work?
And token_key is the fernet key for resource server to encrypt the access token right?

Reply:
It's simply http basic authorization, which is a header provided in the HTTP request such as:

```
Authorization: Basic {BASE64 IDENTITY:PASSWORD}
```

There may be other oauth2 resource server authentication methods besid basic, which is why there is some expansion room. 

And yes, the fernet key is a key specific to this resource server that encrypts tokens for exchange, so that "other resource servers" can't accidentally or maliciously use tokens issued to a different RS.